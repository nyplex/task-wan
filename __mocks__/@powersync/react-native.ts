module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
