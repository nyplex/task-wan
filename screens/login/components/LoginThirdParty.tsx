import { HStack } from "@/components/gluestack/hstack";
import SocialAuthButton from "@/components/buttons/SocialAuthButton";

const LoginThirdParty = () => {
  return (
    <HStack className="items-center justify-around mt-8 w-full">
      <SocialAuthButton provider="google" />
      <SocialAuthButton provider="apple" />
    </HStack>
  );
};

export default LoginThirdParty;
