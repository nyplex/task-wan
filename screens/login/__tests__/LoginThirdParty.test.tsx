import React from "react";
import { render } from "@testing-library/react-native";
import LoginThirdParty from "../components/LoginThirdParty";

describe("LoginThirdParty", () => {
  it("renders two SocialAuthButton components", () => {
    const { getAllByTestId } = render(<LoginThirdParty />);
    // SocialAuthButton should have testID="social-auth-button"
    expect(getAllByTestId("social-auth-button").length).toBe(2);
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(<LoginThirdParty />);
    expect(toJSON()).toMatchSnapshot();
  });
});
