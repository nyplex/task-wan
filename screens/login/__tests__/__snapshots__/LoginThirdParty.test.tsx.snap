// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginThirdParty matches the snapshot 1`] = `
<View
  className="flex-row items-center justify-around mt-8 w-full"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="data-[focus-visible=true]:outline-none data-[focus-visible=true]:ring-indicator-info data-[focus-visible=true]:ring-2 data-[disabled=true]:opacity-40 w-[50px] h-[50px] items-center justify-center rounded-[15px]"
    collapsable={false}
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    style={
      {
        "backgroundColor": "#EEE",
      }
    }
    testID="social-auth-button"
  >
    <View
      className="w-[25px] h-[25px]"
    >
      <ViewManagerAdapter_ExpoImage
        accessibilityLabel="googleIcon"
        containerViewRef={"[React.ref]"}
        contentFit="cover"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height="100%"
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={[]}
        source={
          [
            {
              "testUri": "../../../assets/images/google-logo-light.png",
            },
          ]
        }
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
        testID="social-auth-image"
        transition={null}
        width="100%"
      />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    className="data-[focus-visible=true]:outline-none data-[focus-visible=true]:ring-indicator-info data-[focus-visible=true]:ring-2 data-[disabled=true]:opacity-40 w-[50px] h-[50px] items-center justify-center rounded-[15px]"
    collapsable={false}
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
      }
    }
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    states={
      {
        "active": false,
        "disabled": undefined,
        "focus": false,
        "focusVisible": false,
        "hover": false,
      }
    }
    style={
      {
        "backgroundColor": "#EEE",
      }
    }
    testID="social-auth-button"
  >
    <View
      className="w-[25px] h-[31px]"
    >
      <ViewManagerAdapter_ExpoImage
        accessibilityLabel="appleIcon"
        containerViewRef={"[React.ref]"}
        contentFit="cover"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height="100%"
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={[]}
        source={
          [
            {
              "testUri": "../../../assets/images/apple-logo-light.png",
            },
          ]
        }
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
        testID="social-auth-image"
        transition={null}
        width="100%"
      />
    </View>
  </View>
</View>
`;
